### PR TITLE
Improve test Coverage 

### PR DESCRIPTION
**Pull request by hav-noms**
Remove deprecated code
1. deprecate Proxy.setUseDELEGATECALL 
2. deprecate SynthetixState.setPreferredCurrency

Added tests for improved unit test coverage

### CODE REVIEW

File: contracts/SynthetixState.sol [hunk 1]
```diff
@@ -102,15 +98,10 @@ contract SynthetixState is Owned, State, LimitedSetup, ISynthetixState {
...
-     * @notice Set preferred currency for a user
-     * @dev Only the associated contract may call this.
-     * @param account The account to set the preferred currency for
-     * @param currencyKey The new preferred currency
-     */
-    function setPreferredCurrency(address account, bytes4 currencyKey) external onlyAssociatedContract {
-        preferredCurrency[account] = currencyKey;
-    }
+    // NOTICE: This feature was never used in the system and has been deprecated
+    // function setPreferredCurrency(address account, bytes4 currencyKey) external onlyAssociatedContract {
```
Comment (justinjmoses) [inline contracts/SynthetixState.sol:102]: let's just remove it - no need for commenting it out

File: test/contracts/Depot.js [hunk 2]
```diff
@@ -1115,4 +1234,32 @@ contract('Depot', async accounts => {
...
+
+		beforeEach(async () => {
+			// Send some SNX to the Depot contract
+			await synthetix.transfer(depot.address, snxAmount, {
+				from: owner,
+			});
+		});
+
+		it('when non owner withdrawSynthetix calls then revert', async () => {
+			await assert.revert(
```
Comment (justinjmoses) [inline test/contracts/Depot.js:1249]: please use `helpers.onlyGivenAddressCanInvoke()`

File: test/contracts/Depot.js [hunk 3]
```diff
@@ -778,6 +778,125 @@ contract('Depot', async accounts => {
...
+			const purchaser = address1;
+			const purchaserSynthAmount = toUnit('2000');
+			const depotSNXAmount = toUnit('1000000');
+			const synthsToSend = toUnit('1');
+			let snxToPurchase;
+			let payload;
+			let txn;
+
+			beforeEach(async () => {
+				payload = { from: purchaser };
```
Comment (justinjmoses) [inline test/contracts/Depot.js:844]: `payload` is a bit of a misnomer - how about `fromPurchaser`

File: test/contracts/Depot.js [hunk 4]
```diff
@@ -778,6 +778,125 @@ contract('Depot', async accounts => {
...
+		describe('exchangeEtherForSNXAtRate', () => {
+			const ethToSend = toUnit('1');
+			let snxToPurchase;
+			let payload;
+			let txn;
+
+			beforeEach(async () => {
+				const purchaseValueDollars = multiplyDecimal(ethToSend, ethRate);
+				snxToPurchase = divideDecimal(purchaseValueDollars, snxRate);
+				payload = { from: purchaser, value: ethToSend };
```
Comment (justinjmoses) [inline test/contracts/Depot.js:790]: `payload` => how about `ethToSendFromPurchaser`

File: contracts/SynthetixState.sol (hunk 1 above)
Comment (hav-noms) [reply contracts/SynthetixState.sol:102]: deleted

File: test/contracts/Depot.js (hunk 3 above)
Comment (hav-noms) [reply test/contracts/Depot.js:844]: done

File: test/contracts/Depot.js (hunk 4 above)
Comment (hav-noms) [reply test/contracts/Depot.js]: done as above

File: test/contracts/Depot.js (hunk 2 above)
Comment (hav-noms) [reply test/contracts/Depot.js]: using onlyGivenAddressCanInvoke

**Pull request merged**